### PR TITLE
Fix WorkflowActionsFactory to check both successful and failed builds…

### PR DESCRIPTION
**Pull request by aviral282**
For issue - https://issues.jenkins.io/browse/JENKINS-76148

This pull request improves the logic for collecting HTML report actions in workflow jobs and adds comprehensive unit tests for the `WorkflowActionsFactory`. The main change ensures that build-level reports are collected from the last successful build, while project-level reports are collected from the last build, regardless of its result. Additionally, the new test suite thoroughly verifies the behavior for various job and build scenarios.

**Improvements to report collection logic:**

* Updated `WorkflowActionsFactory.createFor(Job j)` to separately collect build-level HTML reports from the last successful build and project-level reports from the last build, ensuring both types are included appropriately.

### Testing done

* Added a new `WorkflowActionsFactoryTest.java` test suite with multiple tests covering non-workflow jobs, jobs with no builds, jobs with successful and failed builds, and edge cases like null builds and class name matching.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests that demonstrate the feature works or the issue is fixed

### CODE REVIEW

**Comment by aviral282** [comment]
> Thanks very much for the pull request. Please merge with the upstream master branch so that you can include the most recent changes. The plugin has switched from JUnit 4 based tests to JUnit 5. That will require some changes in your automated test.

Hello @MarkEWaite  I have the made the latest changes and they are tested on our end now

**Comment by aviral282** [comment]
Hello @MarkEWaite  Can this be reviewed and merged? we have added tested and rebased with latest changes.

**Review by MarkEWaite — APPROVED** [review]
The changes look good to me.  I'm not a maintainer of this plugin.  I don't have permission to merge the pull request.  That will need to wait for a review and merge by one of the maintainers of the plugin.

File: src/main/java/htmlpublisher/workflow/WorkflowActionsFactory.java [hunk 1]
```diff
@@ -56,22 +56,27 @@ public Collection<? extends Action> createFor(Job j) {
         List<Action> actions = new LinkedList<>();
         if (j.getClass().getCanonicalName().startsWith("org.jenkinsci.plugins.workflow"))
         {
-            final Run<?,?> r = j.getLastSuccessfulBuild();
-            if (r != null) {
-                // If reports are being saved on the build level (keep for all builds)
-                List<HtmlPublisherTarget.HTMLBuildAction> reports = r.getActions(HtmlPublisherTarget.HTMLBuildAction.class);
+            final Run<?,?> lastSuccessful = j.getLastSuccessfulBuild();
+            final Run<?,?> lastBuild = j.getLastBuild();
```
Comment (jtnord) [inline src/main/java/htmlpublisher/workflow/WorkflowActionsFactory.java]: Are you sure you want this and not `lastCompleteBuild`?
`lastBuild` may still be running so reports may not be finalized.

Comment (jglick) [reply src/main/java/htmlpublisher/workflow/WorkflowActionsFactory.java]: Oh I was not aware of a broken link. If true, that is clearly a bug.
